extensions/amp-cxense/0.1/amp-cxense.js

```diff
@@ -0,0 +1,276 @@
+                    self._loaded = true;
+                    return resolve(arg);
+                };
+
+                if (self._target && self._CXV) {
+                    self._CXV.Widgets.get('#' + self._id, function (embed) {
+                        self._embed = embed;
+
+                        if (self._isPlayer) {
+                            let selector = '#' + self._id + '.metaplayer';
+                            self._CXV.Widgets.get(selector, function (mpf) {
+
+                                self._mpf = mpf;
+                                mpf.listen('ready', () => {
+                                    self.applyFillContent(self._target);
+                                    newResolve(self);
+                                });
+                            });
+                        } else {
+                            self.applyFillContent(self._target);
+                            newResolve(self);
+                        }
+                    });
+                } else {
+                    newResolve(self);
+                }
+            });
+        });
+    }
+
+    /** @override */
+    unlayoutCallback() {
+        this._embed && this._embed.destroy();
+        this._target && this._target.parentNode.removeChild(this._target);
+        this._target = null;
+        return true;
+    }
+
+    /** @override */
+    pauseCallback() {
+        if (! this._pauseMpfSync()) {
+            this._pauseMpf();
+        }
+    }
+
+    /** @private */
+    _pauseMpf() {
+        let self = this;
+
+        return new Promise((resolve) => {
+            if (self._target && self._CXV) {
+                self._CXV.Widgets.get('#' + self._target.getAttribute('id') + '.metaplayer', (mpf) => {
+                    self._mpf = self._mpf || mpf;
+                    mpf.video.pause();
+                    resolve();
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    /** @private */
+    _pauseMpfSync() {
+        return !!(this._mpf && (this._mpf.video.pause() || this._mpf.video.paused));
+    }
+
+    /** @private */
+    _buildWidgetPlaceholder() {
+        const doc = this._getDoc();
+
+        let placeholder = doc.createElement('div');
+        placeholder.className = 'amp-cxense-placeholder';
+
+        let spinner = doc.createElement('div');
+        spinner.className = 'amp-mpf-loader';
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+
+        placeholder.appendChild(spinner);
+
+        this.element.appendChild(placeholder);
+        this.applyFillContent(placeholder);
+    }
+
+    /** @private */
+    _injectEmbedScript () {
+        return this._injectScript(this._src);
+    }
+
+    /** @private */
+    _injectScript (src) {
+        const doc = this._getDoc();
+        const script = doc.createElement('script');
```

Comment:
Yes. True. We don't allow external resources to be executed inside the top level window. Only inside the 3p frame. We do prefetch FB SDK for caching purposes and preferably we only load the SDK needed only once on the main context but in FB case there was an issue with only loading it in the main window so we were forced to also load it on every 3p iframe.

Twitter for example only loads its SDK once in master frame, checkout `getTwttr` function in `3p/twitter.js`.